pkg/crc/config/secret_config.go

```diff
@@ -54,30 +47,17 @@ func (c *SecretStorage) Set(key string, value interface{}) error {
 }
 
 func (c *SecretStorage) Unset(key string) error {
-	if !c.storeAccessible {
-		return ErrSecretsNotAccessible
-	}
 	err := keyring.Delete(c.secretService, key)
-	if errors.Is(err, keyring.ErrNotFound) {
+	if err != nil || errors.Is(err, keyring.ErrNotFound) {
+		logging.Debugf("error while getting config from secret store: %v", err)
 		return nil
 	}
 	return err
 }
 
-func keyringAccessible() bool {
```

Comment:
I'm not 100% clear what changes this removal causes? If the keyring is not accessible, instead of caching that it's not accessible, we will be calling the `keyring.*` functions, which will in turn tell us they cannot access teh keyring? So it's just a loss of efficiency, and we no longer have a specific ErrSecretsNotAccessible error?

Reply:
its an attempt to address https://github.com/crc-org/crc/issues/3458 , currently this `keyringAccessible()` check is causing `crc setup` to be stuck when the keychain is not accessible, by removing the check and caching of the result during initialization `crc setup` will be able to proceed and daemon will be started in those scenarios

(looking at it again, i think we should keep the check and the error and move it to body of `keychain.Set`, `keychain.Unset` etc. so its called when we want to access the keychain and if not accessible we get a proper error reported)